pkg/bindinfo/global_handle.go

```diff
@@ -682,3 +685,47 @@ func (h *globalBindingHandle) Stats(_ *variable.SessionVars) (map[string]any, er
 	m[lastPlanBindingUpdateTime] = h.getLastUpdateTime().String()
 	return m, nil
 }
+
+// LoadBindingsFromStorageToCache loads global bindings from storage to the memory cache.
+func (h *globalBindingHandle) LoadBindingsFromStorage(sqlDigest string) (Bindings, error) {
+	if sqlDigest == "" {
+		return nil, nil
+	}
+	bindings, err, _ := h.syncBindingSingleflight.Do(sqlDigest, func() (any, error) {
+		return h.loadBindingsFromStorage(sqlDigest)
+	})
+	if err != nil {
+		logutil.BgLogger().Warn("fail to LoadBindingsFromStorageToCache", zap.Error(err))
+		return nil, err
+	}
+	if bindings == nil {
+		return nil, nil
+	}
+	return bindings.(Bindings), err
+}
+
+func (h *globalBindingHandle) loadBindingsFromStorage(sqlDigest string) (any, error) {
```

Comment:
Why 2 separate functions? It seems like we can merge them into one?

Reply:
Because I don't want to ```h.syncBindingSingleflight.Do``` with a large closure. It is so ugly.